### ISSUE TITLE
Nested queries versus no results

### ISSUE THREAD

**Issue by XoliloX** [tidyverse/dbplyr#1007]
Dear @mgirlich,

I am facing a situation where I either have short SQL queries but no results or a long SQL query that gives results. I am probably doing something wrong (the short version has less aliases but doesn't pick the desired LHS). I looked into #728 and #726 but was not able to find a solution.

When I execute the following code (that has too many aliases), I get results:

```
s <- memdb_frame("col1" = 1:10,"col2" = 2:11,"col9" = 9:18)
t <- memdb_frame("col10" = 10:19,"col11" = 11:20)
c <- memdb_frame("col1" = 1:10,"col12" = 12:21,"col13" = 13:22)
a <- memdb_frame("col14" = 14:23,"col15" = 15:24,"col16" = 16:25)
p <- memdb_frame("col23" = 23:32,"col24" = 24:33,"col25" = 25:34,"col5" = 5:14)

left_join(s,t, 
          by = c("col1" = "col10"), 
          suffix = c("s","t"),
          x_as = "s",
          y_as = "t") %>%
  left_join(c, by = c("col9" = "col1"),
            suffix = c("s","c"),
            x_as = "s",
            y_as = "c") %>% 
  left_join(a, by = c("col11" = "col14"),
            suffix = c("t","a"),
           x_as = "t",
            y_as = "a")%>%
  left_join(p, by = c("col15" = "col23"),
            suffix = c("a","p"),
            x_as = "a",
            y_as = "p") %>% head(5)
```

And `show_query()` prints out:

```
SELECT `a`.*, `col24`, `col25`, `col5`
FROM (
  SELECT `t`.*, `col15`, `col16`
  FROM (
    SELECT `s`.*, `col11`, `col12`, `col13`
    FROM `dbplyr_016` AS `s`
    LEFT JOIN `dbplyr_017` AS `t`
      ON (`s`.`col1` = `t`.`col10`)
    LEFT JOIN `dbplyr_018` AS `c`
      ON (`s`.`col9` = `c`.`col1`)
  ) AS `t`
  LEFT JOIN `dbplyr_019` AS `a`
    ON (`t`.`col11` = `a`.`col14`)
) AS `a`
LEFT JOIN `dbplyr_020` AS `p`
  ON (`a`.`col15` = `p`.`col23`)
```
which is not optimized for the Azure set-up I've been assigned.
So, in order to simplify the query, I pulled away the aliases on LHS from join 2 to 4:

```
left_join(s,t, 
          by = c("col1" = "col10"), 
          suffix = c("s","t"),
          x_as = "s",
          y_as = "t") %>%
  left_join(c, by = c("col9" = "col1"),
            suffix = c("s","c"),
            #x_as = "s",
            y_as = "c") %>% 
  left_join(a, by = c("col11" = "col14"),
            suffix = c("t","a"),
          #x_as = "t",
            y_as = "a")%>%
  left_join(p, by = c("col15" = "col23"),
            suffix = c("a","p"),
            #x_as = "a",
            y_as = "p") %>% show_query()
```

which produces the following SQL query:

```
SELECT
  `s`.*,  `col11`,  `col12`,  `col13`,  `col15`,  `col16`,  `col24`,  `col25`,  `col5`
FROM `dbplyr_016` AS `s`
LEFT JOIN `dbplyr_017` AS `t`
  ON (`s`.`col1` = `t`.`col10`)
LEFT JOIN `dbplyr_018` AS `c`
  ON (`s`.`col9` = `c`.`col1`)
LEFT JOIN `dbplyr_019` AS `a`
  ON (`s`.`col11` = `a`.`col14`)
LEFT JOIN `dbplyr_020` AS `p`
  ON (`s`.`col15` = `p`.`col23`)
```
`collect()`returns: `Error in collect(): ! Failed to collect lazy table. Caused by error: ! no such column: s.col11`.
And indeed, the column `col11` comes from `t` and not `s`. So how can I ensure to have the right LHS picked while having a short SQL query?

In advance, thanks a lot for your answer!

Best,
D.

**Comment by mgirlich** [comment]
Fixed in [fe93ae6](https://github.com/tidyverse/dbplyr/pull/984/commits/fe93ae6026a93327ae03183789a8a3b347d8743a)

**Comment by XoliloX** [comment]
Great !!!! Thanks a lot !!!

Best,
D.
